deploy/eck-stack/charts/eck-kibana/templates/kibana.yaml

```
@@ -12,4 +12,49 @@ metadata:
     {{- end }}
 spec:
   version: {{ required "A Kibana version is required" .Values.version }}
-  {{- toYaml .Values.spec | nindent 2 }}
+  {{- /* 
+    The following templates with 'or' are to allow both .spec.field and .field to be set for backwards 
+    compatibility purposes. See https://github.com/elastic/cloud-on-k8s/pull/8192 for details.
+  */ -}}
+  {{- with or ((.Values.spec).image) (.Values.image) }}
+  image: {{ . }}
+  {{- end }}
+  {{- with .Values.count }}
```

I think we also want this field to be backwards compatible.
```suggestion
  {{- with or ((.Values.spec).count) (.Values.count) }}
```